tests/parser/parser.cpp

```diff
@@ -1417,3 +1417,177 @@ TEST(Parser, parseIds) {
+                            "</math>"
+                            "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">"
+                                "some value in mathml"
+                            "</math>"
+                            "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">"
+                                "extra mathml node"
+                            "</math>"
+                        "</when>"
+                        "<when order=\"3\" id=\"wid\">"
+                            "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">"
+                                "some condition in mathml"
+                            "</math>"
+                            "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">"
+                                "some value in mathml"
+                            "</math>"
+                        "</when>"
+                        "<when order=\"3\" id=\"wid\">"
+                            "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">"
+                                "some condition in mathml"
+                            "</math>"
+                            "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">"
+                                "some value in mathml"
+                            "</math>"
+                        "</when>"
+                    "</reset>"
+                    "<reset variable=\"I_na\" order=\"2\" id=\"rid\">"
+                        "<when order=\"5.9\" goods=\"socks\">"
+                            "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">"
+                                "some condition in mathml"
+                            "</math>"
+                        "</when>"
+                    "</reset>"
+                    "<reset variable=\"I_na\" order=\"2\" id=\"rid\">"
+                        "<when />"
+                    "</reset>"
+                    "<reset id=\"r3id\">"
+                        "<when order=\"\"/>"
+                    "<about>"
+                        "Some description of importance."
+                    "</about>"
+                    "</reset>"
+                    "<reset variable=\"V_k\" order=\"-\" start=\"now\"/>"
+                    "<reset variable=\"variable1\" order=\"0\">"
+                        "non empty whitespace."
+                        "<when order=\"1\">"
+                            "illegal content."
+                            "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">"
+                                "some condition in mathml"
+                            "</math>"
+                            "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">"
+                                "some value in mathml"
+                            "</math>"
+                            "<variable/>"
+                        "</when>"
+                    "</reset>"
+                "</component>"
+            "</model>";
+
+    const std::vector<std::string> expectedErrors = {
```

Comment:
As discussed above, it might be nice to handle the case where there is no variable name (or order value) as such. Also, I can't recall whether we discussed it before (and whether it would be easy to do here), but it might be worth specifying the line/column where an error was found?